### PR TITLE
Add doc for notification API

### PR DESCRIPTION
**Pull request by cgx027**
Add doc for notification API. 

**TODO** replace github link after on-http PR # 410 merge.

### CODE REVIEW

File: docs/rackhd/notification_api.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
+Notification
+-------------------
+
+The notification API provides functionality for sending notifications to RackHD
+from a node. For example, a node could send notification to RackHD to inform that
+OS install has finished.
```
Comment (yyscamper) [inline docs/rackhd/notification_api.rst]: Use noun, install -> installation

File: docs/rackhd/notification_api.rst [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+The notification API is only available from the south bound.
```
Comment (yyscamper) [inline docs/rackhd/notification_api.rst]: south bound --> southbound

File: docs/rackhd/notification_api.rst [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+How does it work
+~~~~~~~~~~~~~~~~~~~~~~~
+
+When a node calls a notification API, the RackHD on-http process will get acknowledged
+and then send a AMQP message to a exchange named 'on.events', with routing key set to
```
Comment (yyscamper) [inline docs/rackhd/notification_api.rst]: a -> an, set -> setting

File: docs/rackhd/notification_api.rst [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+'notification' or 'notification.<id>' depending on the parameters sent along when
+calling the notification API.
+
+Any tasks running in on-taskgraph process that is expecting a notification will need
```
Comment (yyscamper) [inline docs/rackhd/notification_api.rst]: `any tasks` -> `any task`

**Comment by heckj** [comment]
@cgx027 love the diagram! Thank you

👍 with Felix's language update suggestions and replacing with the relevant code once https://github.com/rackHD/on-http/pull/410 has been merged

**Comment by heckj** [comment]
:+1:

File: docs/rackhd/notification_api.rst [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+Use notification API in OS installation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A typical OS installation needs two notifications. The first one notifies that OS has been installed
+to the disk on the target node. The second one notifies that the OS has been successfully booted
+on the target node.
+
+The first notificatioin is typically sent in the 'postinstall' section of the kickstart file. 
+For example (**Todo**: replace this link after on-http PR #410 merges):
```
Comment (yyscamper) [inline docs/rackhd/notification_api.rst]: remove todo

**Comment by yyscamper** [comment]
:+1: after removing the todo.

**Pull request merged**